### PR TITLE
Fix data reviews URL

### LINKED ISSUE #564: Monthly Data Reviews Link Missing
The link to monthly data reviews on the timeline doesn't exist anymore. https://github.com/python-discord/site/blob/93be3f40a07f988d63cacb2dc24ca35c17fee76a/pydis_site/templates/home/timeline.html#L114

### PR DESCRIPTION
**Pull request by ethansocal**
I updated the URL on the timeline for data reviews from `https://pythondiscord.com/pages/data-reviews/` to `https://www.notion.so/6784e3a9752444e89d19e65fd4510d8d?v=69268794b9d2454a9c28e1550e2923c2`.

### CODE REVIEW

**Comment by ethansocal** [comment]
Resolves #564

File: pydis_site/templates/home/timeline.html [hunk 1]
```diff
@@ -111,7 +111,7 @@ <h2>The privacy policy is created</h2>
             pretty much as soon as our new bot and site starts collecting some data. To this day, we keep <a
                 href="https://pythondiscord.com/pages/privacy/">our privacy policy</a> up to date with all
             changes, and since April 2020 we've started doing <a
-                href="https://pythondiscord.com/pages/data-reviews/">monthly data reviews</a>.</p>
+                href="https://www.notion.so/6784e3a9752444e89d19e65fd4510d8d?v=69268794b9d2454a9c28e1550e2923c2">monthly data reviews</a>.</p>
```
Comment (Akarys42) [inline pydis_site/templates/home/timeline.html]:
```suggestion
                href="https://pythondiscord.notion.site/6784e3a9752444e89d19e65fd4510d8d">monthly data reviews</a>.</p>
```

**Review by Akarys42 — DISMISSED** [review]
Hey @ethansocal, thanks a lot for the PR! After internal discussions, we decided to instead use the `notion.site` link to avoid domain blocks. Sorry 😅

Feel free to apply directly from GitHub or from your computer!

**Review by wookie184 — APPROVED** [review]
Thanks!

**Review by Bluenix2 — APPROVED** [review]
Great! Thank you for your contribution

**Review by Akarys42 — APPROVED** [review]
Perfect, thank you!

**Pull request merged**